SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -146,30 +166,82 @@ void USpatialSender::GainAuthorityThenAddComponent(USpatialActorChannel* Channel
 			// adding the subobject.
 			PendingSubobjectAttachment->PendingAuthorityDelegations.Add(ComponentId);
 			Receiver->PendingEntitySubobjectDelegations.Add(
-				MakeTuple(static_cast<Worker_EntityId_Key>(Channel->GetEntityId()), ComponentId),
+				MakeTuple(static_cast<Worker_EntityId_Key>(EntityId), ComponentId),
 				PendingSubobjectAttachment);
 
-			EntityACL->ComponentWriteAcl.Add(Info->SchemaComponents[Type], AuthoritativeWorkerRequirementSet);
+			NewComponentIds.Add(ComponentId);
 		}
 	});
 
-	FWorkerComponentUpdate Update = EntityACL->CreateEntityAclUpdate();
+	// If this worker is EntityACL authoritative, we can directly update the component IDs to gain authority over.
+	if (StaticComponentView->HasAuthority(EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
+	{
+		const FClassInfo& ActorInfo = ClassInfoManager->GetOrCreateClassInfoByClass(Channel->Actor->GetClass());
+		const WorkerAttributeSet WorkerAttribute{ ActorInfo.WorkerType.ToString() };
```

Comment:
Nit. Would prefer for consistency.
```suggestion
		const WorkerAttributeSet WorkerAttribute = { ActorInfo.WorkerType.ToString() };
```